### DISCUSSION TITLE
Inaccurate results from entities

### DISCUSSION THREAD

**Discussion by ghost** [speechly/speechly#301] [Q&A] [answered]
Hello everyone,

I'm working with a speech recognition tool called Speechly and trying to create a template for recognizing product codes. The template should be able to identify a product code that consists of a letter followed by a two-digit number (e.g., A19, B42, etc.) and output it as "product id [code]".

I've defined the following variables in my template:

letter_var = [a | b | c | ... | x | y | z]
code_var = [00..99]
product_code_var = $letter_var $code_var
product_var = [product ID]
And I'm using the following code to extract the product code:
*_ $product_var(product) $product_code_var(product_code)

However, when I test the template with some sample data, I'm getting unexpected results. For example, when I say "product id S14", the output I get is "product id product id s one product_code four IDS1". It seems like the template is adding an extra "id" and not recognizing the number "14" correctly.

I've included screenshots of my template code, the expanded utterances from the Speechly dashboard, and the user data outputs that show the mismatches between the expected and actual outputs.

My desired output is "product id A00..Z99", where the letter can be any uppercase letter and the number can be any two-digit number. Can you please help me understand what's causing the issue and how I can fix it? Is it a problem with the speech recognition model, the template itself, or something else?

Thank you for your help.

<img width="1272" alt="1" src="https://user-images.githubusercontent.com/47230031/233437584-e9d84b19-6fa9-4e14-b294-7fe114086793.png">
<img width="587" alt="2" src="https://user-images.githubusercontent.com/47230031/233437586-93b114d5-7454-4c18-b346-dc61e0f21d14.png">
<img width="539" alt="3" src="https://user-images.githubusercontent.com/47230031/233437590-57868abc-802a-4238-8683-2d33fc9729f7.png">

**Comment by bigdatabaracus** [comment]
Thank you @delegatepattern  for reaching out. I think @ge2918st might be able to help you here. Let him get back to you.

**Comment by ge2918st** [comment, marked as answer]
Hi @delegatepattern thanks for using Speechly. We had a look into your configuration and we found out a number of reasons why this was not working as expected. We tried to refactor your configuration based on your expectations and desired outcomes. You may have a look below, copy-paste in Speechly DB, redeploy your application, and test it out 👍 :

```
# LETTERS
letter_seq = [a | b | c | ... | x | y | z]

# NUMBERS
num_seq = [0..99]

# INTENTS
product = [
  *product product id $letter_seq(letter_seq) $num_seq(num_seq)
  ]

 intent = [
    $product
    ]

$intent
```

As for the capitalisation of the letter values ('letter_seq'), there's currently no difference between upper and lower case and the ASR output is by default with lower case.

If you have any more questions happy to help you out.

**Reply by ghost** [reply]
Thank you so much George, worked nicely! 

I have a question if you don't mind:

I wanted to extend my appreciation for your valuable expertise. I have a question regarding the language configuration of the ASR system. Is it possible to utilize Spanish as the primary spoken language?

Moreover, I am curious if the training method for text-based adaptation would remain consistent in this context.

Thank you for your kind assistance.

**Comment by ge2918st** [comment]
Hi @delegatepattern. Thanks for using Speechly. We're glad we could help you previously and you got your configuration performing as wanted. 

Your last question about transcription in other languages (other than English): Speechly provides several language speech recognition models in a range of sizes and capacities including Spanish, but not for live-streaming technology (`https://docs.speechly.com/features/models`). Speechly can analyze large volumes of data in several languages asynchronously if you have pre-recorded audio or video footage that needs a transcript. Speechly ASR models support only the English language though.

Are you interested specifically in live-streaming technology (ASR transcription) or pre-recorded audio (Batch transcription)?

Thanks,
George

**Reply by ghost** [reply]
Dear George,

Thank you for your prompt response and clarifying the capabilities of Speechly's ASR system. I appreciate your assistance in the past, which helped me configure the system according to my requirements.

Regarding my previous question about transcription in languages other than English, I am specifically interested in live-streaming technology. I aim to utilize ASR to convert real-time speech into text and fill forms dynamically. Unfortunately, I understand that Speechly's ASR models currently do not support live-streaming for languages other than English.

I also came across the Whisper model during my research. While it does not support streaming, I discovered that it limits certain Speechly features when used. Considering this limitation, I wanted to ask if you have any suggestions or alternative approaches that could help me achieve real-time data entry through voice commands during livestreaming.

I watched the video titled "Improve CRM Data Quality With Voice User Interfaces" on your YouTube channel, where you demonstrate a similar use case. I was impressed by the seamless integration of voice commands into form filling. I would like to achieve a similar functionality with data entry based on voice commands during livestreaming.

I must say that I really like Speechly and would prefer to continue using it for my clients. The SDK is user-friendly and provides a great experience compared to other SDKs that can be cumbersome. Considering my interest in the Spanish language, I would appreciate any suggestions or guidance from you on how I can leverage Speechly for Spanish language support.

Thank you once again for your support and assistance. I look forward to your response.

video link: https://www.youtube.com/watch?v=6GcgPcMOuQk&t=7s

**Comment by ge2918st** [comment]
Hi @delegatepattern! 

We appreciate your kind words. Thanks 🙏 When users tell us how much value their usage of our product has given them, we are thrilled! 

The Speechly team is aiming to increase the number of languages accessible for ASR technology, and Spanish will be added shortly. 

When it's ready for testing, stay tuned for a launch notice that will be sent out 👍 

Thanks,
George
